### DISCUSSION TITLE
flink-cdc在全量阶段使用了SELECT * FROM TABLE读取数据。

### DISCUSSION THREAD

**Discussion by banyue0618** [apache/flink-cdc#3058] [Q&A]
flink-cdc读取数据第一阶段，快照读，采用了select * 读取数据。
请问下应该怎么配置，只读取我需要的字段呢。因为：我的一张表有两百个字段，但是实际上我只需要其中的五个字段，完全没有必要采用星号。
看flink-cdc的源码。直接代码中写死了星号。
![image](https://github.com/ververica/flink-cdc-connectors/assets/40993873/b412bef3-9d1c-481f-90b9-a04e60f4cea1)
目前已经进行的尝试是配置debezium属性，`debezium.column.include.list`,但是根据数据库查询日志，还是使用的select *
然后我查看debezium的源码，在构建projection的时候校验了`isColumnsFiltered`，根据这个属性来决定select的字段信息
![image](https://github.com/ververica/flink-cdc-connectors/assets/40993873/5827e8ac-7fb4-47c6-9fa2-bb7613240ff6)
![image](https://github.com/ververica/flink-cdc-connectors/assets/40993873/eae2db88-2d5a-4751-a4ef-ce3ded85907c)

为什么flink-cdc没有这样做呢。
感谢您的帮助。

**Comment by belveyf** [comment]
看源码Debezium读取时候使用select *， column.include.list参数只是在生成change event时候使用。
可以参考下MySqlSnapshotChangeEventSource类
